mlflow/pyfunc/__init__.py

```diff
@@ -3001,25 +3005,17 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
                             _PythonModelPyfuncWrapper(python_model, None, None),
                         )
                     except Exception as e:
-                        if infer_signature_from_type_hints:
-                            _logger.warning(
-                                "Failed to infer model signature from type hints or input example. "
-                                f"Inferring model signature from input example failure: {e}",
-                            )
-                        else:
-                            _logger.warning(
-                                "Failed to infer model signature: "
-                                f"Type hint {type_hints} cannot be used to infer model signature."
-                                f"Inferring model signature from input example failure: {e}"
-                            )
+                        _logger.warning(
+                            f"Failed to infer model signature from input example, error: {e}",
+                        )
                 else:
-                    # if infer_signature_from_type_hints, warnings are emitted
+                    # if signature is inferred from type hints, warnings are emitted
                     # in _infer_signature_from_type_hints
-                    if not infer_signature_from_type_hints:
+                    if not should_infer_signature_from_type_hints:
                         _logger.warning(
                             "Failed to infer model signature: "
-                            f"Type hint {type_hints} cannot be used to infer model signature."
-                            "Input example is not provided, model signature cannot be inferred."
+                            f"Type hint {type_hints} cannot be used to infer model signature, "
+                            "input example is not provided, model signature cannot be inferred."
```

Comment:
nit: I guess article and conjunction is missing? Shall we say `and no input example is provided?`